flake.nix

```diff
@@ -9,32 +9,24 @@
     };
   };
 
-  outputs = { self, nixpkgs, home-manager, flake-utils }:
-    {
-      overlays = import ./overlays;
-    } //
-    flake-utils.lib.eachSystem [
-      "aarch64-linux"
-      "i686-linux"
-    ]
-      (system:
-        let pkgs = import nixpkgs {
-          inherit system;
-          overlays = self.overlays;
-        }; in
-        rec {
-          lib = {
-            nix-on-droid = { config }: import ./modules {
-              inherit pkgs config;
-              isFlake = true;
-              home-manager = (import home-manager { });
-            };
-          };
+  outputs = { self, nixpkgs, home-manager, flake-utils }: let
+    supportedSystems = [ "aarch64-linux" "i686-linux" ];
+  in flake-utils.lib.eachSystem supportedSystems (system: let
+    defaultPkgs = import nixpkgs {
+      inherit system;
+      overlays = [ self.overlay ];
+    };
+  in rec {
+    lib.nix-on-droid = { pkgs ? defaultPkgs, config }: import ./modules {
```

Comment:
Not necessary for this PR, but I would say, it is reasonable to add `home-manager` as argument.

Reply:
Since `home-manager` now only provides the path and not any configuration, I don't see a need to add it here. Using native methods to override inputs in Flakes feels better to me.

Reply:
But you can not apply patches to the input natively with flakes but you could do it programmatically.